packages/migrate/src/Migrate.ts

```diff
@@ -55,44 +57,95 @@ export class Migrate {
     return this.engine.reset()
   }
 
-  public createMigration(params: EngineArgs.CreateMigrationInput): Promise<EngineResults.CreateMigrationOutput> {
-    return this.engine.createMigration(params)
+  public async createMigration(
+    params: Omit<EngineArgs.CreateMigrationInput, 'migrationsList'>,
+  ): Promise<{ generatedMigrationName: string | undefined }> {
+    if (!this.migrationsDirectoryPath) throw new Error('this.migrationsDirectoryPath is undefined')
+
+    const migrationsList = await listMigrations(this.migrationsDirectoryPath)
+    const { connectorType, generatedMigrationName, extension, migrationScript } = await this.engine.createMigration({
```

Comment:
❓ As `listMigrations` does not throw or abort on file read errors all the error handling for broken files etc is done on the Rust side of things, correct? Does this make our lifes significantly easier or actually harder as we have to serialize all the errors back and forth? 🤔

Reply:
Life for us is unchanged. `listMigrations` treats loads most errors lazily, rather than eagerly, so that we don't need to change too many things at once. Our life is neither observably harder nor easier.